lib/server/db/model/user.js

```diff
@@ -1,14 +1,33 @@
-var mongoose = require('mongoose');
+var mongoose  = require('mongoose'),
+    validator = require('node-mongoose-validator');
+
 mongoose.connect('mongodb://localhost/beans');
 
-var User = mongoose.model(
-    'User',
-    {
-        username: String,
-        password: String,
-        friends: []
-    }
-);
+// define a schema
+var userSchema = new mongoose.Schema({
+    firstName: { type: String, required: true },
+    lastName:  { type: String, required: true },
+    username:  { type: String, required: true },
+    password:  { type: String, required: true }
+
+});
+
+// Validations
+userSchema.path('firstName').validate(validator.notEmpty(), 'Please provide a first name.');
```

Comment:
[Document#validate](http://mongoosejs.com/docs/api.html#document_Document-validate) takes a function as it's second argument.

Our function should take in an `err` parameter. If the `err` value is truthy, we should log out the error to the console.